src/movedescriptor.js

```diff
@@ -104,6 +104,20 @@ MoveDescriptor.prototype.toString = function() {
 	return result;
 };
 
+MoveDescriptor.prototype.toUCIString = function() {
+	var result = bt.squareToString(this._from);
+	if(this.isCastling() && bt.fileFromSquare(this._from) !== 'e') {
+		// UCI castling is KxR of same color for chess960 by convention
+		result += bt.squareToString(this._optionalSquare1);
```

Comment:
I'm still not convinced by this KxR convention. The UCI protocol reference available on Shredderchess.com (https://www.shredderchess.com/download/div/uci.zip) describe this convention as applicable only to Chess960 games, while classic form (e1g1) is supposed to be used for standard chess games. I guess that some chess engine (typically Stockfish) may support both the classic and the KxR form, but chess engines that do not implement Chess960 are likely to support only the classic form.

I've also find the following discussion, that makes me think that using KxR notation in standard chess games may not be a good idea:
https://github.com/niklasf/shakmaty/issues/13

For these reasons, I've implemented UCI move generation as follows in the master:
- classic form ('e1g1') is used to generate castling moves for standard games,
- KxR is used for Chess960 games,
- a optional flag is provided to force KxR for standard games, but this behavior is disabled by default.

For parsing, both forms are accepted for standard games, while KxR is mandatory for Chess960.